### DISCUSSION TITLE
fail to push submodules with HTTPS

### DISCUSSION THREAD

**Discussion by egoltzman** [community/community#61670] [New to GitHub]
I have a repo with some AWS submodules and when I try to issue:
**git push --recurse-submodules=on-demand**
I get this:
**Pushing submodule 'lib/AWS/Device-Shadow-for-AWS-IoT-embedded-sdk'**
Here I get a popup window:
![image](https://github.com/community/community/assets/3929373/94fea8d5-f84f-424e-ac7a-2181afc46a1e)
And I select "Sign in with a code", I get the code and visit the URL it point me to and I issue the device activation successfully and press the "Authorize git-ecosystem" button, I get "Congratulations, you're all set!" and then this:
**remote: Permission to aws/Device-Shadow-for-AWS-IoT-embedded-sdk.git denied to XYZ.
fatal: unable to access 'https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk.git/': The requested URL returned error: 403
Unable to push submodule 'lib/AWS/Device-Shadow-for-AWS-IoT-embedded-sdk'**
How can I resolve this?
I follow some suggestions I saw to remove the current credentials from Windows credential manager as well as use GitHUB CLI to set the HTTP authentication login, I did those with no change in the above behavior.

**Comment by w8ste** [comment]
Hi,
is it important to you to use HTTPS ? 

If not i would recommend you to use ssh, since once set up it is a lot easier to use then HTTPS. If you want to check out this [doc](https://docs.github.com/en/authentication/connecting-to-github-with-ssh)

Hope this helps :)
